server/routes/api/documents.ts

```diff
@@ -795,82 +801,132 @@ router.post("documents.search_titles", auth(), pagination(), async (ctx) => {
+    } else {
+      if (!user) {
+        throw AuthenticationError("Authentication error");
+      }
 
-  if (userId) {
-    assertUuid(userId, "userId must be a UUID");
-    collaboratorIds = [userId];
-  }
+      teamId = user.teamId;
 
-  if (dateFilter) {
-    assertIn(
-      dateFilter,
-      ["day", "week", "month", "year"],
-      "dateFilter must be one of day,week,month,year"
-    );
-  }
+      if (collectionId) {
+        assertUuid(collectionId, "collectionId must be a UUID");
+        const collection = await Collection.scope({
+          method: ["withMembership", user.id],
+        }).findByPk(collectionId);
+        authorize(user, "read", collection);
+      }
 
-  const { results, totalCount } = await Document.searchForUser(user, query, {
-    includeArchived: includeArchived === "true",
-    includeDrafts: includeDrafts === "true",
-    collaboratorIds,
-    collectionId,
-    dateFilter,
-    offset,
-    limit,
-  });
+      let collaboratorIds = undefined;
 
-  const documents = results.map((result) => result.document);
+      if (userId) {
+        assertUuid(userId, "userId must be a UUID");
+        collaboratorIds = [userId];
+      }
 
-  const data = await Promise.all(
-    results.map(async (result) => {
-      const document = await presentDocument(result.document);
-      return { ...result, document };
-    })
-  );
+      if (dateFilter) {
+        assertIn(
+          dateFilter,
+          ["day", "week", "month", "year"],
+          "dateFilter must be one of day,week,month,year"
+        );
+      }
 
-  // When requesting subsequent pages of search results we don't want to record
-  // duplicate search query records
-  if (offset === 0) {
-    SearchQuery.create({
-      userId: user.id,
-      teamId: user.teamId,
-      source: ctx.state.authType,
-      query,
-      results: totalCount,
-    });
-  }
+      response = await Document.searchForUser(user, query, {
+        includeArchived: includeArchived === "true",
+        includeDrafts: includeDrafts === "true",
+        collaboratorIds,
+        collectionId,
+        dateFilter,
+        offset,
+        limit,
+        minWords,
+        maxWords,
+      });
+    }
 
-  const policies = presentPolicies(user, documents);
+    const { results, totalCount } = response;
+    const documents = results.map((result) => result.document);
 
-  ctx.body = {
-    pagination: ctx.state.pagination,
-    data,
-    policies,
-  };
-});
+    const data = await Promise.all(
+      results.map(async (result) => {
+        const document = await presentDocument(result.document);
+        return { ...result, document };
+      })
+    );
+
+    // When requesting subsequent pages of search results we don't want to record
+    // duplicate search query records
+    if (offset === 0) {
+      SearchQuery.create({
+        userId: user?.id,
+        teamId,
+        shareId,
+        source: ctx.state.authType || "app", // we need to write a valid enum member into here
```

Comment:
Is there not a valid `authType` because we're potentially using a shareId. If that's the case it might be better to fix this in the auth middleware with a new enum, `unauthenticated` or similar?

Reply:
I think defaulting to "app" is correct, because if we're actually interested in the "source" of the search query. If it's not via the API, then we should assume it's through the app, as no api calls are possible without an api token.

That it's unauthenticated matters to conceptually to authtype but that's not what's being stored semantically. It just happens to line up